operator/pkg/ciliumidentity/controller_test.go

```diff
@@ -197,3 +204,253 @@ func verifyCIDUsageInCES(ctx context.Context, fakeClient *k8sClient.FakeClientse
+				{},
+			},
+			cidUpdateList: []*capi_v2.CiliumIdentity{
+				nil,
+				cid2,
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+				{cid2.SecurityLabels},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			cid, _, ns, fakeClient, _, h := initHiveTest(true)
+			ctx := context.Background()
+			tlog := hivetest.Logger(t)
+			if err := h.Start(tlog, ctx); err != nil {
+				t.Fatalf("starting hive encountered an error: %s", err)
+			}
+			defer func() {
+				if err := h.Stop(tlog, ctx); err != nil {
+					t.Fatalf("stopping hive encountered an error: %v", err)
+				}
+			}()
+
+			nsStore, err := (*ns).Store(ctx)
+			if err != nil {
+				t.Fatalf("get namespace store: %v", err)
+			}
+			if err := nsStore.CacheStore().Add(ns1); err != nil {
+				t.Fatalf("create namespace: %v", err)
+			}
+
+			cidStore, _ := (*cid).Store(ctx)
+
+			fakeClient.CiliumFakeClientset.PrependReactor("create", "ciliumidentities", func(action k8sTesting.Action) (handled bool, ret runtime.Object, err error) {
+				pa := action.(k8sTesting.CreateAction)
+				cid, _ := pa.GetObject().(*capi_v2.CiliumIdentity)
+				cidStore.CacheStore().Add(cid)
+				return true, nil, nil
+			})
+			fakeClient.CiliumFakeClientset.PrependReactor("update", "ciliumidentities", func(action k8sTesting.Action) (handled bool, ret runtime.Object, err error) {
+				pa := action.(k8sTesting.UpdateAction)
+				cid, _ := pa.GetObject().(*capi_v2.CiliumIdentity)
+				cidStore.CacheStore().Update(cid)
+				return true, nil, nil
+			})
+			fakeClient.CiliumFakeClientset.PrependReactor("delete", "ciliumidentities", func(action k8sTesting.Action) (handled bool, ret runtime.Object, err error) {
+				pa := action.(k8sTesting.DeleteAction)
+				cidName := pa.GetName()
+				cid, _, _ := cidStore.GetByKey(cidResourceKey(cidName))
+				cidStore.CacheStore().Delete(cid)
+				return true, nil, nil
+			})
+
+			// Test starts here.
+			// Run podActions in the sequence defined in the test case.
+			// Verify CID state after each podAction.
+			for actionIdx, podAction := range tc.podActions {
+				pod := podAction.podObj
+				switch podAction.action {
+				case "create":
+					if _, err := fakeClient.Slim().CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{}); err != nil {
+						t.Fatalf("create pod: %v", err)
+					}
+				case "update":
+					if _, err := fakeClient.Slim().CoreV1().Pods(pod.Namespace).Update(ctx, pod, metav1.UpdateOptions{}); err != nil {
+						t.Fatalf("update pod: %v", err)
+					}
+				case "delete":
+					if err := fakeClient.Slim().CoreV1().Pods(pod.Namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{}); err != nil {
+						t.Fatalf("delete pod: %v", err)
+					}
+				}
+
+				if len(tc.cidUpdateList) > 0 && tc.cidUpdateList[actionIdx] != nil {
+					cidObjList := cidStore.CacheStore().List()
+					if len(cidObjList) == 0 {
+						t.Fatalf("listed CIDs = 0, want >= 1")
```

Comment:
Fatalf calls FailNow which marks the function as having failed and stops its execution by calling runtime.Goexit(which then runs all deferred calls in the current goroutine). Execution will continue at the next test or benchmark.

I think we should remove the below return.

Reply:
Done.